### PR TITLE
Add instructions for package installer on Ubuntu/Deb

### PR DESCRIPTION
**Pull request by dzimine**
While this is WIP, I push it to master and deploy to https://docs.stackstorm.com/latest to easily test / refer / comment. 

Once complete, we cherry-pick this PR to 1.3 docs

### CODE REVIEW

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/stackstorm/p1454453003004444

File: docs/source/install/deb.rst [hunk 1]
```diff
@@ -1,84 +1,210 @@
...
+    sudo apt-get install -y mongodb-server rabbitmq-server postgresql
+
+
+Setup repositories
+~~~~~~~~~~~~~~~~~~~
+
+  .. code-block:: bash
+
+    wget -qO - https://bintray.com/user/downloadSubjectPublicKey?username=bintray | sudo apt-key add -
+    echo "deb https://dl.bintray.com/stackstorm/`lsb_release -cs`_staging stable main" | sudo tee /etc/apt/sources.list.d/st2-stable.list
```
Comment (armab) [inline docs/source/install/deb.rst]: @lakshmi-kannan Is it a plan to let users install from `staging` repos until creating Staging st2 box with all CI that deploys packages to normal production repos?

File: docs/source/install/deb.rst [hunk 2]
```diff
@@ -1,84 +1,210 @@
...
 
-   WOWZERS! A **deprecation** notice! Please note that this documentation and associated DEB packages are undergoing review and maintenance. Our core packages for Debian and Ubuntu are undergoing a complete overhaul to incorporate `dh-virtualenv <http://dh-virtualenv.readthedocs.org>`_ into our base packages. As such, please consider these instructions **DEPRECATED** for the time being. Hold tight, these packages and docs are coming very soon.
+* Take these steps on all boxes where you run stackstorm remote actions, **including** ``localhost``.
 
-   If you are feeling adventurous and want to attempt an installation using this method, we have left this documentation here for you to better understand the internals of the system as it currently exists
+  .. code-block:: bash
 
-   In the meantime, please feel free to take a look at many of our installation methods, including our `All-in-one Installer </install/all_in_one.rst>`_.
+    # Create an SSH system user
+    useradd stanley
```
Comment (armab) [inline docs/source/install/deb.rst]: BTW https://github.com/StackStorm/st2-packages/issues/117

File: docs/source/install/deb.rst (hunk 1 above)
Comment (dzimine) [inline docs/source/install/deb.rst]: yes, exactly. (verified with @lakshmi-kannan )

File: docs/source/install/deb.rst (hunk 2 above)
Comment (dzimine) [inline docs/source/install/deb.rst]: ack, decided to make it user / installer-script job. Use creating is fine (and when I follow the doc on the local box, it tells me that the user is created, and safely ignored, but note that exactly the same set applies to the remote boxes).

File: docs/source/install/deb.rst [hunk 3]
```diff
@@ -1,84 +1,210 @@
...
+    mkdir -p /home/stanley/.ssh
+    chmod 0700 /home/stanley/.ssh
 
+    # Generate ssh keys on StackStorm box and copy over public key into remote box.
+    ssh-keygen -f /home/stanley/.ssh/stanley_rsa -P ""
+    cp ${KEY_LOCATION}/stanley_rsa.pub /home/stanley/.ssh/stanley_rsa.pub
```
Comment (lakshmi-kannan) [inline docs/source/install/deb.rst]: This line is not needed. ssh-keygen generates public keys in the right place already.

```
root@pkgtestingbox:/home/vagrant# ssh-keygen -f /home/stanley/.ssh/stanley_rsa -P ""
Generating public/private rsa key pair.
Your identification has been saved in /home/stanley/.ssh/stanley_rsa.
Your public key has been saved in /home/stanley/.ssh/stanley_rsa.pub
```

File: docs/source/install/deb.rst [hunk 4]
```diff
@@ -1,84 +1,210 @@
...
-and dependencies:
+    echo "stanley    ALL=(ALL)       NOPASSWD: SETENV: ALL" >> /etc/sudoers.d/st2
 
-::
+* Adjust configuration in ``/etc/st2/st2.conf``:
 
-  deb http://downloads.stackstorm.net/deb/ trusty_unstable main
+  .. sourcecode:: ini
 
+    [system_user]
```
Comment (lakshmi-kannan) [inline docs/source/install/deb.rst]: Default st2.conf ships with the right configuration.

File: docs/source/install/deb.rst [hunk 5]
```diff
@@ -1,84 +1,210 @@
...
-    apt-get install -y ${aptlist}
+    st2 -h
 
-Pip dependencies
-''''''''''''''''
+    st2 action list --pack=core
 
-Use :github_st2:`requirements.txt </requirements.txt>` file from the release brunch of st2 source.
+    # List the actions from a 'core' pack
+    st2 action list --pack=core
```
Comment (lakshmi-kannan) [inline docs/source/install/deb.rst]: I still have auth enabled. This is not working. Documentation is missing a step. 

```
vagrant@pkgtestingbox:~$ grep -A 5 "auth" /etc/st2/st2.conf
[auth]
host = 0.0.0.0
port = 9100
use_ssl = False
debug = False
enable = True
logging = /etc/st2/logging.auth.conf
```

Comment (dzimine) [inline docs/source/install/deb.rst]: Looks like it is set here. https://github.com/StackStorm/st2-packages/blob/master/scripts/generate_st2_config.sh#L26 Why this doesn't work, or is it been called?

Comment (lakshmi-kannan) [inline docs/source/install/deb.rst]: I don't think that is even invoked anywhere. https://github.com/StackStorm/st2/pull/2466 might do it. I am not sure.

Comment (lakshmi-kannan) [inline docs/source/install/deb.rst]: vagrant@st2dev /mnt/src/storm/st2-packages/packages/st2 (master●)$ grep "generate" Makefile                                                                                                                                                 [ruby-1.9.3p484]
vagrant@st2dev /mnt/src/storm/st2-packages/packages/st2 (master●)$

Comment (lakshmi-kannan) [inline docs/source/install/deb.rst]: Looks like the generate_st2_config.sh script is only used with rspec tests.

Comment (armab) [inline docs/source/install/deb.rst]: Right, that script is of Rspec tests.

I see you already found where to edit default `st2.conf`: https://github.com/StackStorm/st2/pull/2466

**Pull request merged**